File: src/services/orcamentoService.ts
import api from "./api";
import { Decimal } from "decimal.js";

export interface Orcamento {
  id: number;
  nomeCliente: string;
  telefone: string;
  dataDoOrcamento: string;
  origem: string;
  destino: string;
  distancia: string;
  valorTotal: number;
  paradas: string;
  tipoViagemOrcamento?: string;
  descricaoIdaOrcamento?: string;
  descricaoVoltaOrcamento?: string;
  textoGerado?: string;
  custoDistancia?: number;
  custoPedagios?: number;
  custoCombustivel?: number;
  custoMotorista?: number;
}

export interface CadastroOrcamentoData {
  nomeCliente: string;
  telefone: string;
  origem: string;
  destino: string;
  distancia: string;
  paradas: string;
  valorTotal: number;
  tipoViagemOrcamento?: string;
  descricaoIdaOrcamento?: string;
  descricaoVoltaOrcamento?: string;
  textoGerado?: string;
}

const ROTA = "/orcamentos";

export const orcamentoService = {
  async listar(): Promise<Orcamento[]> {
    const res = await api.get<Orcamento[]>(ROTA);
    return res.data;
  },

  async salvar(dados: CadastroOrcamentoData): Promise<Orcamento> {
    const res = await api.post<Orcamento>(ROTA, {
      ...dados,
      valorTotal: new Decimal(dados.valorTotal).toString(),
    });
    return res.data;
  },

  excluir: async (id: number): Promise<void> => {
    await api.delete(`/orcamentos/${id}`);
  },
};

File: backend/ubuntu@52.15.79.19/src/main/java/br/com/rafas/transportes/api/service/OrcamentoService.java
/**
 * @author falvesmac
 */

package br.com.rafas.transportes.api.service;

import br.com.rafas.transportes.api.domain.Orcamento;
import br.com.rafas.transportes.api.dto.DadosCadastroOrcamento;
import br.com.rafas.transportes.api.dto.DadosDetalhamentoOrcamento;
import br.com.rafas.transportes.api.repository.OrcamentoRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.time.LocalDate;
import java.time.Period;
import java.util.List;

@Service
public class OrcamentoService {

  @Autowired
  private OrcamentoRepository repository;

  public Orcamento salvar(DadosCadastroOrcamento dados) {
    var orcamento = new Orcamento();

    orcamento.setNomeCliente(dados.nomeCliente());
    orcamento.setTelefone(dados.telefone());
    orcamento.setOrigem(dados.origem());
    orcamento.setDestino(dados.destino());
    orcamento.setDistancia(dados.distancia());
    orcamento.setParadas(dados.paradas());
    orcamento.setValorTotal(dados.valorTotal());

    orcamento.setDataDoOrcamento(LocalDate.now());

    orcamento.setTipoViagemOrcamento(dados.tipoViagemOrcamento());
    orcamento.setDescricaoIdaOrcamento(dados.descricaoIdaOrcamento());
    orcamento.setDescricaoVoltaOrcamento(dados.descricaoVoltaOrcamento());

    repository.save(orcamento);
    return orcamento;
  }

  @Transactional(readOnly = true)
  public List<DadosDetalhamentoOrcamento> listarTodos() {
    return repository.findAll().stream()
            .map(DadosDetalhamentoOrcamento::new)
            .toList();
  }

  @Transactional
  public void excluir(Long id) {
    repository.deleteById(id);
  }

  @Scheduled(cron = "0 0 2 * * *")
  @Transactional
  public void excluirOrcamentosAntigos() {
    LocalDate dataLimite = LocalDate.now().minus(Period.ofDays(60));
    List<Orcamento> orcamentosAntigos = repository.findByDataDoOrcamentoBefore(dataLimite);

    if (!orcamentosAntigos.isEmpty()) {
      repository.deleteAll(orcamentosAntigos);
      System.out.println("Excluídos " + orcamentosAntigos.size() + " orçamentos com mais de 60 dias.");
    }
  }
}